### PR TITLE
mgr: add per pool scrub commands

### PR DESCRIPTION
**Pull request by xiexingguo**
Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by xiexingguo** [comment]
retest this please

File: src/mgr/DaemonServer.cc [hunk 1]
```diff
@@ -1087,6 +1087,76 @@ bool DaemonServer::_handle_command(
...
+    cmd_getval(g_ceph_context, cmdctx->cmdmap, "who", pool_names);
+    if (pool_names.empty()) {
+      ss << "must specify one or more pool names";
+      cmdctx->reply(-EINVAL, ss);
+      return true;
+    }
+    epoch_t epoch;
+    map<int32_t, vector<pg_t>> pgs_by_primary; // legacy
+    map<int32_t, vector<spg_t>> spgs_by_primary;
+    cluster_state.with_osdmap([&](const OSDMap& osdmap) {
```
Comment (liewegas) [inline src/mgr/DaemonServer.cc]: r =

File: src/mgr/DaemonServer.cc [hunk 2]
```diff
@@ -1087,6 +1087,76 @@ bool DaemonServer::_handle_command(
...
+        }
+      }
+    });
+    if (r < 0) {
+      cmdctx->reply(r, ss);
+      return true;
+    }
+    vector<string> keywords;
+    get_str_vec(prefix, keywords);
+    auto& op = keywords.back();
```
Comment (liewegas) [inline src/mgr/DaemonServer.cc]: can we skip this and just change ``op == ...`` below to ``prefix == ...`` instead?

**Comment by xiexingguo** [comment]
http://pulpito.ceph.com/xxg-2019-02-21_02:56:41-rados-wip-pool-scrub-distro-basic-smithi/

**Pull request merged**